Formula/instead.rb

```diff
@@ -29,7 +29,7 @@ def install
       system "cmake", "..", "-DWITH_GTK2=OFF",
                             "-DWITH_LUAJIT=ON",
                             "-DLUA_INCLUDE_DIR=#{luajit.opt_include}/luajit-2.1",
-                            "-DLUA_LIBRARY=#{luajit.opt_lib}/#{shared_library("libluajit")}",
+                            "-DLUA_LIBRARY=#{luajit.opt_lib}/#{shared_library("libluajit-5.1")}",
```

Comment:
Shouldn't this be or even get the version from the formula?
```suggestion
                            "-DLUA_LIBRARY=#{luajit.opt_lib}/#{shared_library("libluajit","5.1")}",
```

Reply:
This is very confusing, but the version used here is the the version of Lua used by the JIT (5.1), not of `luajit`.  I forgot to mention that the reason this is even necessary is because the Linux bottle for `luajit-openresty` doesn't have an unversioned `libluajit.so`, while the macOS bottle does have an unversioned `libluajit.dylib`.  

I should have checked the `luajit-openresty` formula earlier because I can see now that we manually create the unversioned `libluajit.dylib` symlink on macOS, but it is hardcoded to for `libluajit.dylib` rather than `shared_library("libluajit")`.  This is a mistake that is trivial to fix, and doing so should mean this entire PR is unnecessary.  I will close this and fix the issue with `luajit-openresty` instead, and we should be able to bottle this with no changes needed.